spark/src/main/antlr4/io/delta/sql/parser/DeltaSqlBase.g4

```diff
@@ -73,6 +73,7 @@ singleStatement
 // If you add keywords here that should not be reserved, add them to 'nonReserved' list.
 statement
     : VACUUM (path=STRING | table=qualifiedName)
+        (USING INVENTORY (reservoirPath=STRING | reservoirTable=qualifiedName |  reservoirQuery=statement))?
```

Comment:
Can we put the `statement` in parentheses? The RETAIN etc. follow this, and it may end up being ambiguous at the end of a SQL query.

Another question: what is the definition of "reservoirPath"? Is it a path to a Delta table? I think allowing paths makes sense for commands like VACUUM that only operate on Delta tables, but as inputs it's not a given that the input is even a Delta table. Maybe we can leave that out and just rely on the regular:
```
delta.`/path/to/table`
```
(How do you write backticks in markdown code blocks? :) )

Reply:
I tried surrounding the query with LEFT_PAREN and RIGHT_PAREN symbols 
`statement
    : VACUUM (path=STRING | table=qualifiedName)
        (USING INVENTORY (reservoirPath=STRING | reservoirTable=qualifiedName |  LEFT_PAREN reservoirQuery=statement RIGHT_PAREN))?
        (RETAIN number HOURS)? (DRY RUN)?
`

However, I ran into EOF error for some statements that are supposed to go to delegated parser
`DELETE FROM testTable WHERE id IN (0,10,20,30,40)` with stacktrace like 

> == SQL ==
>[info] DELETE FROM testTable WHERE id IN (0,10,20,30,40)
>[info] ------------------------------------------------^^^
>[info]   at org.apache.spark.sql.catalyst.parser.ParseException.withCommand(parsers.scala:257)
>[info]   at io.delta.sql.parser.DeltaSqlParser.parse(DeltaSqlParser.scala:128)
>[info]   at io.delta.sql.parser.DeltaSqlParser.parsePlan(DeltaSqlParser.scala:77)
>[info]   at org.apache.spark.sql.SparkSession.$anonfun$sql$5(SparkSession.scala:684)
>[info]   at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:138)
>[info]   at org.apache.spark.sql.SparkSession.$anonfun$sql$4(SparkSession.scala:683)
>[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
>[info]   at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:682)
>[info]   at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:713)


I am still trying to see how I can fix this (If you have any pointers pls share :) ), but pushed the initial commit without the parenthesis

Reply:
regarding 
> delta.`/path/to/table`

I think if we use `reservoirTable=qualifiedName` it covers both table names and `format`. `paths` We have a logic that raises exception if the passed table is not delta. I think we can remove `reservoirPath=STRING` from the grammar and just use table or query statement alone. What do you think

Reply:
> I think if we use reservoirTable=qualifiedName it covers both table names and format. paths We have a logic that raises exception if the passed table is not delta. I think we can remove reservoirPath=STRING from the grammar and just use table (this also covers delta.path/to/table) or a query statement alone. What do you think

+1!

Reply:
> I am still trying to see how I can fix this (If you have any pointers pls share :) ), but pushed the initial commit without the parenthesis
You seem to have managed to fix it though, is that right?

Reply:
yes, managed to get it fixed.